OpenSim/Simulation/Model/PhysicalFrame.cpp

```diff
@@ -40,36 +40,40 @@ using namespace OpenSim;
 /**
 * Default constructor.
 */
-RigidFrame::RigidFrame() : Frame()
+PhysicalFrame::PhysicalFrame() : Frame()
 {
-    setNull();
-
+    setAuthors("Matt DeMers");
 }
 
-
-void RigidFrame::setNull()
+const SimTK::MobilizedBody& PhysicalFrame::getMobilizedBody() const
 {
-    setAuthors("Matt DeMers");
+    return getModel().getMatterSubsystem().getMobilizedBody(_mbIndex);
 }
 
-const SimTK::MobilizedBody& RigidFrame::getMobilizedBody() const
+SimTK::MobilizedBody& PhysicalFrame::updMobilizedBody() 
 {
-    return getModel().getMatterSubsystem().getMobilizedBody(_index);
+    return updModel().updMatterSubsystem().updMobilizedBody(_mbIndex);
 }
 
-SimTK::MobilizedBody& RigidFrame::updMobilizedBody() 
+/*
+* Implementation of Frame interface by PhysicalFrame.
+* 
+*/
+const SimTK::Transform& PhysicalFrame::
+    calcGroundTransform(const SimTK::State& s) const
 {
-    return updModel().updMatterSubsystem().updMobilizedBody(_index);
+    // return X_GF = X_GB * X_BF;
+    return getMobilizedBody().getBodyTransform(s);
 }
 
-/**
-* Implementation of Frame interface by RigidFrame
-*/
-SimTK::Transform RigidFrame::calcGroundTransform(const SimTK::State& state) const {
+void PhysicalFrame::extendAddToSystem(SimTK::MultibodySystem& system) const
+{
+}
 
-    const SimTK::MobilizedBody &B = getModel().getMatterSubsystem().getMobilizedBody(_index);
-    const SimTK::Transform& X_GB = B.getBodyTransform(state);
 
-    return X_GB*getTransformInMobilizedBody();
+SimTK::Transform PhysicalFrame::extendFindTransformInBaseFrame() const
+{
+    Transform identity;
+    return identity.setToZero();
```

Comment:
You can just write `return Transform();`. They are defined to initialize to identity.